### PR TITLE
Remove dynamic config

### LINKED ISSUE #3662: Remove dynamic configuration from Agent
as discussed previously dynamic configuration is being removed.

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Remove dynamic configuration from docs, code and anywhere else

#### Which issue(s) this PR fixes

Fixes #3662 

#### PR Checklist

- [X] CHANGELOG updated
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ internal API changes are not present.
 Main (unreleased)
 -----------------
 
+### Breaking changes
+
+- Removal of dynamic configuration. (@mattdurham)
```
Comment (rfratto) [inline CHANGELOG.md:15]:
```suggestion
- The experimental dynamic configuration feature has been removed in favor of Flow mode. (@mattdurham)
```

File: tools/smoke/internal/smoke.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func New(logger log.Logger, cfg Config) (*Smoke, error) {
 			w.WriteHeader(http.StatusOK)
 		}
 		if cfg.SimulateErrors {
-			rand.Seed(time.Now().UnixNano())
+			rand.New(rand.NewSource(1)).Seed(time.Now().UnixNano())
```
Comment (rfratto) [inline tools/smoke/internal/smoke.go:89]: This doesn't seem right, you're creating a new random number generator, seeding it, but then never using it anywhere.

File: CHANGELOG.md (hunk 1 above)
Comment (rfratto) [inline CHANGELOG.md:15]: Can you add this to the upgrade guide too?

File: tools/smoke/internal/smoke.go (hunk 2 above)
Comment (mattdurham) [reply tools/smoke/internal/smoke.go]: Your right. The code is deprecated in 1.20, added a nolint check for it for the time being.

File: docs/sources/static/upgrade-guide.md [hunk 3]
```diff
@@ -19,6 +19,11 @@ static mode.
 > [upgrade-guide-operator]: {{< relref "../operator/upgrade-guide.md" >}}
 > [upgrade-guide-flow]: {{< relref "../flow/upgrade-guide.md" >}}
 
+### Removal of Dynamic Configuration
+
+The experimental feature Dynamic Configuration has been removed. The use case of dynamic configuration will be replaced 
+with Modules in Grafana Agent Flow.
```
Comment (rfratto) [inline docs/sources/static/upgrade-guide.md:25]: Link to Modules?

**Review by rfratto — APPROVED** [review]

**Pull request merged**